### ISSUE TITLE
Migration of COMPASS files away from oceans11

### ISSUE THREAD

**Issue by pwolfram** [MPAS-Dev/MPAS-Model#294]
Oceans11 will be going away and we should migrate data files for COMPASS inputs to a more stable source.  LANL IC /usr/projects has been proposed, but there may be other options.

The purpose of this issue is to:

 1. Determine where the files should be stored
 2. Make sure that we migrate COMPASS to use these files in their new location, at which point we can close out this issue. 

cc @mark-petersen, @sbrus89, @maltrud, @philipwjones, @xylar

**Comment by pwolfram** [comment]
To kick-start the discussion:

I'm fine with LANL IC but don't like that it is only here at LANL.  External collaborators can't get in and this limits open-development.  

We do now have official Google Drive support -- we could also places files there and then have the option to give external collaborators access to these files as needed.  However, I'm not sure there is a clean way to access these files from the command line, but maybe this isn't a substantial problem since a local version of the COMPASS inputs is all that is really needed and someone would need to build this themselves (it just wouldn't be automated via COMPASS to populate the database).

**Comment by pwolfram** [comment]
Here is a list of the affected files: https://gist.github.com/975b77aa479556cd32030c0cea6793d5

Note, @akturner and @matthewhoffman  should also engage here since there are some seaice and landice files that also need migrated.

**Comment by xylar** [comment]
Oceans11 has been "going away" for *years*.  Is there evidence that it's going away for real?

Anyway, I've been talking with Rick Wagner (one of the E3SM Infrastructure team members) and we're going to likely move our diagnostics data (the obs, mapping files, etc. used by MPAS-Analysis and E3SM_Daigs) to petrel at Argonne.  That machine has a super fast connection to our other machines via Globus (or easy https access just like Oceans11) and seems like a promising option for COMPASS data as well.  I'll post more once I have a chance to try it out myself.

**Comment by pwolfram** [comment]
@maltrud and @philipwjones don't have root anymore, which is a great reason to move away from it... I'm not personally interested in putting files there anymore.

Thanks for looking into the petrel option at Argonne @xylar!

Perhaps we should be thinking about redundant, multiple locations for access.  Mirroring can be tricky.  But, given globus as an option this will make things much easier though, so I really like this option.

**Comment by mark-petersen** [comment]
I would vote for moving all data for MPAS (for LANL) to the E3SM IC repo:
https://web.lcrc.anl.gov/public/e3sm/

We thought of this last year, and made these directories:
https://web.lcrc.anl.gov/public/e3sm/mpas_standalonedata/mpas-albany-landice/
https://web.lcrc.anl.gov/public/e3sm/mpas_standalonedata/mpas-ocean/
https://web.lcrc.anl.gov/public/e3sm/mpas_standalonedata/mpas-seaice/
https://web.lcrc.anl.gov/public/e3sm/mpas_standalonedata/DEMSI/

Advantages:

  1.  Same firewall access as needed for E3SM.
  2.  Easy to access. Just copy your files to a www directory at nersc (but I can’t find path right now)
  3.  Can use same download process as E3SM.

**Comment by philipwjones** [comment]
I gave the machine away a couple years ago due to the increasing demands for administration of a public-facing machine and the new policies that permitted use of easier and more robust cloud services. At that time I asked everyone to find a new home and move their data elsewhere and most have done so. All of the old repos were moved and most of the input data. The current owners of the machine have been graciously continuing to serve our remaining data (doesn't require much additional effort above what they need the machine for), but could choose to decommission the machine or decide not to serve that data at any time.

**Comment by xylar** [comment]
Thanks @philipwjones, that's helpful information.

@mark-petersen, the LCRC hosting is from Anvil (Blues), not NERSC.  But yes, that's the other option at Argonne beside Petrel.  The advantage is that many of us already have access.  The down side is that we can't add new people very easily, whereas Petrel is independent and meant specifically for file hosting so it would allow us to add people (e.g. NCAR collaborators) who could upload data sets.

**Comment by xylar** [comment]
It seems like Anvil is the easiest solution.  I can start working on this in August if no one gets to it sooner.

**Comment by xylar** [comment]
This has become critical and needs to happen in the next few weeks.

**Comment by maltrud** [comment]
@xylar , @mark-petersen : is everything in /raid/mpas_data?

**Comment by mark-petersen** [comment]
Yes. I'm copying it over to a ccs scratch drive right now. In the end it needs to be transferred to Anvil (or somewhere) where it can be downloaded. mpas_data is 39GB total.

**Comment by maltrud** [comment]
i just copied it to poseidon

**Comment by mark-petersen** [comment]
For safe keeping I put it in two locations:
```
ccscs1:/scratch/mpeterse/oceans11_backup_200114
ar-tn:/campaign/w20_ocean_time_step/oceans11_backup_200114
```
I'll copy to Anvil this week.

**Comment by xylar** [comment]
It seems like a good opportunity to copy only the files that COMPASS actually makes use of, so I would only move over the files that we need rather than all 39 GB if there is an appreciable number of files we don't need.

**Comment by mark-petersen** [comment]
Yes, I thought the same thing. We are all tight on time, but that is a great way to start.

**Comment by mark-petersen** [comment]
We can see the files that are listed in COMPASS that access oceans11 with these commands:
```
cd testing_and_setup/compass
git grep -B 1 'oceans11'|grep file_name
```
and see the subdirectories with:
```
git grep -B 1 'oceans11'|grep file_name|grep landice
git grep -B 1 'oceans11'|grep file_name|grep mesh_database
git grep -B 1 'oceans11'|grep file_name|grep initial_condition_database
```

The output looks like this: 
https://gist.githubusercontent.com/mark-petersen/4d0fe39bc6d95590e1a42d9cf26ef318/raw/ef86aeb5984135e5478a027159d09c2091b09a1e/gistfile1.txt

**Comment by xylar** [comment]
@mark-petersen, if you can put all the files on Anvil somewhere temporarily (not in the public server), I will be willing to move only the files currently used into the appropriate place within https://web.lcrc.anl.gov/public/e3sm/mpas_standalonedata/mpas-ocean/

**Comment by xylar** [comment]
Never mind.  I'll do this now.

**Comment by xylar** [comment]
@mark-petersen, the land-ice test cases were taken care of about a year and a half ago!  (Thanks for being on the ball, @matthewhoffman!)

https://github.com/MPAS-Dev/MPAS-Model/pull/97

I don't see any evidence that MPAS-Seaice ever used oceans11 in its test cases (which are separate from COMPASS in any case).

**Comment by xylar** [comment]
Addressed by #429
